q0_linac.py

```diff
@@ -40,38 +40,43 @@ def load_data(self):
         self.heater_runs: List[q0_utils.HeaterRun] = []
 
         with open(self.cryomodule.calib_data_file, "r+") as f:
-            all_data: Dict = json.load(f)
-            data: Dict = all_data[self.time_stamp]
-
-            for heater_run_data in data.values():
-                run = q0_utils.HeaterRun(heater_run_data["Desired Heat Load"])
-                run._start_time = datetime.strptime(
-                    heater_run_data[q0_utils.JSON_START_KEY],
-                    q0_utils.DATETIME_FORMATTER,
-                )
-                run._end_time = datetime.strptime(
-                    heater_run_data[q0_utils.JSON_END_KEY], q0_utils.DATETIME_FORMATTER
-                )
+            with open(self.cryomodule.selected_calib_data_file, "r+") as sf:
+                all_data: Dict = json.load(f)
+                selected_data: Dict = json.load(sf)
+                if self.time_stamp in selected_data:
+                    data: Dict = selected_data[self.time_stamp]
```

Comment:
Ok let me think about this a little bit. Does this mean that once I've created the newly edited data, it's always gonna be what's pulled when I open the GUI?

Reply:
Yeah, it'll only read the edited data. Maybe can do it so that there's a way to select between edited or entire data? Otherwise can leave it so that it'll only plot the original, and someone can manually access the edited when needed.

Reply:
Yeah maybe what we should do instead is try to make this an instance only thing. Like, we load the data into a buffer used by the gui and _that_ gets updated every time we either choose a new calibration/rf measurement (in which case we read from the files) and when we change a selected region. Side note, I just realized that I can expand a region beyond the first and last data points in a run. Is it possible to bound that?

Reply:
Sure that works. So, we're updating a buffer when something new (either region or measurement) is selected, but the edited measurement json is still being written to? 

I should be able to bound the regions, I'll double check that.

Reply:
If we do this buffer I don't think we need an edited file? Feel free to correct me but my first thought is that we shouldn't need it. And ideally all calculations and plots should be updated